libs/diag/diagnostic.go

```diff
@@ -30,6 +30,16 @@ type Diagnostic struct {
 	ID ID
 }
 
+// Error creates a new error diagnostic.
+func Error_(msg string) Diagnostics {
```

Comment:
NewError maybe?

Reply:
that would be to different from Errorf, I was hoping to keep the similarity there,

Reply:
We have `diag.Warn` and `diag.Warnf`, the severity being `Warning`.

We could do `diag.Err` and `diag.Errf`,

Reply:
Renaming diag.Err to diag.Errf would be very noisy. For now I removed Error_, we can discuss it separately. (I'd prefer to rename severity levels, prefix them with something).